llvm/lib/CodeGen/MachineScheduler.cpp

```diff
@@ -477,7 +477,7 @@ bool MachineScheduler::runOnMachineFunction(MachineFunction &mf) {
 }
 
 bool PostMachineScheduler::runOnMachineFunction(MachineFunction &mf) {
-  if (skipFunction(mf.getFunction()))
+  if (mf.hasFakeUses() || skipFunction(mf.getFunction()))
```

Comment:
That seems like a good reason to keep it. The disabling of post RA scheduling for -fextend-lifetimes is a fairly old feature for our downstream branch, and since then enough has changed (particularly the instruction-referencing LiveDebugValues implementation) that post-RA scheduling no longer appears to cause as many problems as before. I'll investigate to see whether enabling post-RA scheduling with -fextend-lifetimes downstream causes any major debug info regressions, which would necessitate modifying post-RA-sched to respect fake uses in some way, but I suspect that this is no longer needed.

Reply:
Although actually this looks like another existing bug with optnone. in AMDGPU we add a dedicated pass to recognize hazards at -O0 to replace the postRA scheduler hazard handling. That won't happen for higher opt levels for an optnone function